gridded_etl_tools/util_funcs/reindexing.py

```diff
@@ -0,0 +1,22 @@
+import pandas as pd
+import xarray as xr
+
+
+def fill_in_missing_time_steps(dataset: xr.Dataset, freq: str, time_dim: str = "time") -> xr.Dataset:
```

Comment:
Does this still trigger expensive shuffling operations? If so we should put a warning on this function.